multiraft/multiraft.go

```diff
@@ -409,16 +409,15 @@ func (s *state) start() {
 			return
 
 		case req := <-s.reqChan:
-			_, groupOK := s.groups[req.GroupID]
-			_, nodeOK := s.nodes[NodeID(req.Message.From)]
-			groupOK = groupOK || req.Message.Type == raftpb.MsgHeartbeat
-			if !groupOK || !nodeOK || NodeID(req.Message.To) != s.nodeID {
-				log.V(4).Infof("node %v: discarding ill-addressed message %s", s.nodeID, req.GroupID,
-					raft.DescribeMessage(req.Message, s.EntryFormatter))
-				break
-			}
 			log.V(5).Infof("node %v: group %v got message %s", s.nodeID, req.GroupID,
 				raft.DescribeMessage(req.Message, s.EntryFormatter))
+			if _, ok := s.groups[req.GroupID]; !ok {
```

Comment:
Do you not care anymore about not knowing about the node which the message was sent from?